test/client_ui_test.py

```diff
@@ -84,6 +85,74 @@ def test_draw():
     assert f.last == line.ljust(79) + "\n"
 
 
+def test_json_format():
+    fake_time = FakeTime()
+    f = FakeFile()
+
+    # Size is unknown at this point.
+    pb = client.ProgressBar(
+        phase="setting up", output=f, format="json", now=fake_time)
+    line_data = {
+        "transferred": 0,
+        "elapsed": 0.0,
+        "description": "setting up",
+    }
+    line = f'{json.dumps(line_data)}'
+    assert f.last == line + "\n"
```

Comment:
Maybe instead of creating json and comparing the output - which is fragile if the
json writer changes keys order, test parsing the output:

    pb = client.ProgressBar(
        phase="setting up", output=f, format="json", now=fake_time)
    assert json.loads(f.last) == {
        "transferred": 0,
        "elapsed": 0.0,
        "description": "setting up",
    }
    assert f.last.endswith("\n")
    
Benefits:
- test does not depend on key order which is not part of the API
- test does not replicate the tested code
- no need for extra temporary variables (line_dict, line)
- we test what users will do with the output